lib/main.js

```diff
@@ -1,7 +1,23 @@
-import Vue from 'vue'
-import App from './components/app.vue'
+import Vue from 'vue';
+import axios from 'axios';
 
-new Vue({ // eslint-disable-line no-new
+import App from './components/app.vue';
+const globalComponents = [
+  require('./components/error-message.vue'),
```

Comment:
Could we not mix the two import module syntaxes? `import` and `require` do the same thing. I'm in on adopting the new `import` syntax and would replace those `require` expressions to imports.

Reply:
I also generally prefer `import` to `require()`. However, as far as I can tell, `import` can't be used in an array like this. Separately importing the global components then adding them to the array is slightly more verbose than using `require()` in the array. I also don't actually need to give any of the `require()`-d components a name, which I think `import` would require. Do you think it's alright to use `require()` just in this array, or do you think there's a better approach?

Reply:
I understand what you say but I see three major problems with this approach. 

- ‘require’, as opposed to ‘import’, is not (and will not be) supported by EcmaScript. By using ‘require’ you’re forcing this code to be transpiled, even if browsers start to understand ‘import’ statements which is the standard mechanism for loading modules in EcmaScript.
- Dynamic module loading is considered harmful in a web app context because it makes bundling more difficult. Browserify/Webpack can’t know what modules you’re ‘require’-ing and this impacts negatively in your options for building your deliverables.
- As a design guideline, it’s easier to understand a rule without exceptions like “always use import, no matter what”. Also, this way, all modules are imported at the header of each file, which is easier to read, as opposed to having modules being loaded in unexpected places.

I understand that you need to write some more lines but I think it’s worth the effort.

Reply:
Sounds good, I have made that change!